test/core/pipeline/test_pipeline_breakpoints_rag_hybrid.py

```diff
@@ -0,0 +1,153 @@
+import os
+import sys
+
+import pytest
+
+from haystack.components.builders.answer_builder import AnswerBuilder
+from haystack.components.builders.prompt_builder import PromptBuilder
+from haystack.components.embedders import SentenceTransformersDocumentEmbedder, SentenceTransformersTextEmbedder
+from haystack.components.generators import OpenAIGenerator
+from haystack.components.joiners import DocumentJoiner
+from haystack.components.rankers import TransformersSimilarityRanker
+from haystack.components.retrievers.in_memory import InMemoryBM25Retriever, InMemoryEmbeddingRetriever
+from haystack.components.writers import DocumentWriter
+from haystack.document_stores.in_memory import InMemoryDocumentStore
+from haystack.document_stores.types import DuplicatePolicy
+from haystack import Document
+
+from haystack_experimental.core.errors import PipelineBreakpointException
+from haystack_experimental.core.pipeline.pipeline import Pipeline
+
+
+class TestPipelineBreakpoints:
+    """
+    This class contains tests for pipelines with breakpoints.
+    """
+
+    @pytest.fixture
+    def document_store(self):
+        """Create and populate a document store for testing."""
+        documents = [
+            Document(content="My name is Jean and I live in Paris."),
+            Document(content="My name is Mark and I live in Berlin."),
+            Document(content="My name is Giorgio and I live in Rome."),
+        ]
+
+        document_store = InMemoryDocumentStore()
+        doc_writer = DocumentWriter(document_store=document_store, policy=DuplicatePolicy.SKIP)
+        doc_embedder = SentenceTransformersDocumentEmbedder(
```

Comment:
We should mock the TransformersSimilarityRanker, SentenceTransformersDocumentEmbedder and SentenceTransformersTextEmbedder so that they do not download models and that they do not require the sentence-transformers dependency and also do not rquire transformers with torch.

Reply:
Similarly, we should mock the OpenAIChatGenerator and OpenAIGenerator so that tests can be much faster